Watching for file changes with StatReloader
<div>
    <label for="id_name">Name:</label>



<input type="text" name="name" value="testadf" required id="id_name">
    
</div>

  <div>
    <label for="id_price">Price:</label>



<input type="number" name="price" value="9" step="any" required id="id_price">
    
</div>

  <div>
    <label for="id_category">Category:</label>



<input type="text" name="category" value="cat1" required id="id_category">
    
</div>

  <div>
    <label for="id_buyer_address">Buyer address:</label>



<input type="text" name="buyer_address" value="e, e, e 8888" required id="id_buyer_address">
    
      
    
</div>
Internal Server Error: /page2/
Traceback (most recent call last):
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/site_app/views.py", line 68, in page2
    p = Product(name=product_form.cleaned_data['name'], price=product_form.cleaned_data['price'], category=product_form.cleaned_data['category'], owner=request.user, buyer_address=product_form.cleaned_data['buyer_address'], is_bought=False)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/db/models/base.py", line 543, in __init__
    _setattr(self, field.name, rel_obj)
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/db/models/fields/related_descriptors.py", line 287, in __set__
    raise ValueError(
ValueError: Cannot assign "'e, e, e 8888'": "Product.buyer_address" must be a "Address" instance.
/home/gareth/school/software_eng/group-project/site_app/views.py changed, reloading.
Watching for file changes with StatReloader
<QuerySet []>
<QuerySet []>
<div>
    <label for="id_name">Name:</label>



<input type="text" name="name" value="testasdf" required id="id_name">
    
</div>

  <div>
    <label for="id_price">Price:</label>



<input type="number" name="price" value="4" step="any" required id="id_price">
    
</div>

  <div>
    <label for="id_category">Category:</label>



<input type="text" name="category" value="cat1" required id="id_category">
    
</div>

  <div>
    <label for="id_buyer_address">Buyer address:</label>



<input type="text" name="buyer_address" value="e, e, e 8888" required id="id_buyer_address">
    
      
    
</div>
Internal Server Error: /page2/
Traceback (most recent call last):
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/site_app/views.py", line 70, in page2
    line1 = product_form['buyer_address'].split(',')[0]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'BoundField' object has no attribute 'split'
/home/gareth/school/software_eng/group-project/site_app/views.py changed, reloading.
Watching for file changes with StatReloader
<div>
    <label for="id_name">Name:</label>



<input type="text" name="name" value="testasdf" required id="id_name">
    
</div>

  <div>
    <label for="id_price">Price:</label>



<input type="number" name="price" value="4" step="any" required id="id_price">
    
</div>

  <div>
    <label for="id_category">Category:</label>



<input type="text" name="category" value="cat1" required id="id_category">
    
</div>

  <div>
    <label for="id_buyer_address">Buyer address:</label>



<input type="text" name="buyer_address" value="e, e, e 8888" required id="id_buyer_address">
    
      
    
</div>
<QuerySet [<Product: testasdf>]>
Internal Server Error: /
Traceback (most recent call last):
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/site_app/views.py", line 59, in home
    return render(request, 'site_app/index.html', {'products': products})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'buy_product' not found. 'buy_product' is not a valid view function or pattern name.
/home/gareth/school/software_eng/group-project/site_app/views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /
Traceback (most recent call last):
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/site_app/views.py", line 60, in home
    return render(request, 'site_app/index.html', {'products': products})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'buy_product' not found. 'buy_product' is not a valid view function or pattern name.
/home/gareth/school/software_eng/group-project/site_app/views.py changed, reloading.
Watching for file changes with StatReloader
/home/gareth/school/software_eng/group-project/site_app/views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /
Traceback (most recent call last):
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/site_app/views.py", line 61, in home
    return render(request, 'site_app/index.html', {'products': product_list})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/backends/django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/template/defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gareth/school/software_eng/group-project/env/lib/python3.12/site-packages/django/urls/resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'buy_product' not found. 'buy_product' is not a valid view function or pattern name.
<div>
    <label for="id_name">Name:</label>



<input type="text" name="name" value="rob&#x27;s test product" required id="id_name">
    
</div>

  <div>
    <label for="id_price">Price:</label>



<input type="number" name="price" value="4" step="any" required id="id_price">
    
</div>

  <div>
    <label for="id_category">Category:</label>



<input type="text" name="category" value="test category" required id="id_category">
    
</div>

  <div>
    <label for="id_buyer_address">Buyer address:</label>



<input type="text" name="buyer_address" value="aa, ss, ss 88888" required id="id_buyer_address">
    
      
    
</div>
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Watching for file changes with StatReloader
Not Found: /favicon.ico
Watching for file changes with StatReloader
Not Found: /favicon.ico
Watching for file changes with StatReloader
Not Found: /favicon.ico
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\trading_treasure\settings.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /home/
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\site_app\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\site_app\views.py changed, reloading.
Watching for file changes with StatReloader
Forbidden (CSRF token from POST incorrect.): /signup
Forbidden (CSRF token from POST incorrect.): /signup
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\site_app\views.py changed, reloading.
Watching for file changes with StatReloader
Forbidden (CSRF token from POST incorrect.): /signup
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\site_app\views.py changed, reloading.
Watching for file changes with StatReloader
Forbidden (CSRF token from POST incorrect.): /signup
C:\Users\nikki\OneDrive\Desktop\Programming\Git\Trading\CSCI4830-TradingTreasure\site_app\views.py changed, reloading.
Watching for file changes with StatReloader
